core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -1513,8 +1512,22 @@ func (ec *EthConfirmer) ResumePendingTaskRuns(ctx context.Context, head *evmtype
 		return err
 	}
 
+	if len(receipts) > 0 {
+		ec.lggr.Debugf("Resuming %d task runs pending receipt", len(receipts))
+	} else {
+		ec.lggr.Trace("No task runs to resume")
+	}
 	for _, data := range receipts {
-		if err := ec.resumeCallback(data.ID, data.Receipt, nil); err != nil {
+		var err error
+		var output interface{}
+		if data.FailOnRevert && data.Receipt.Status == 0 {
+			err = errors.Errorf("transaction %s reverted on-chain", data.Receipt.TxHash)
+		} else {
+			output = data.Receipt
+		}
+
+		ec.lggr.Tracew("Callback: resuming ethtx with receipt", "output", output, "err", err, "pipelineTaskRunID", data.ID)
+		if err := ec.resumeCallback(data.ID, output, err); err != nil {
```

Comment:
This looks really odd (passing `err` and returning `error`)..
This even triggers govet warning: `shadow: declaration of "err" shadows declaration at line 1521 (govet)`